source/server/active_tcp_socket.cc

```diff
@@ -75,6 +75,22 @@ void ActiveTcpSocket::continueFilterChain(bool success) {
     bool no_error = true;
     if (iter_ == accept_filters_.end()) {
       iter_ = accept_filters_.begin();
+      // TODO(soulxu) This is for PoC, but we should initialize and reset file event
+      // for each filter, then we can back-compatible with existing filter. After all
+      // filter switch to the new method to get peek data, then we can change to this way.
+      if (inspect_buffer_size_ > 0) {
+        listener_filter_buffer_ = std::make_unique<Network::ListenerFilterBufferImpl>(
+            socket_->ioHandle(), listener_.dispatcher(), [this]() { continueFilterChain(false); },
+            [this]() {
+              Network::FilterStatus status = (*iter_)->onData(*listener_filter_buffer_);
+              if (status == Network::FilterStatus::StopIteration) {
+                return;
+              }
+              continueFilterChain(true);
+            },
+            inspect_buffer_size_);
+        listener_filter_buffer_->initialize();
```

Comment:
I think you can move initialize() into the constructor?

Reply:
Yes, but I think I need to separate the PR, then I need this method to initialize and reset the fileevent for iterate each filter. Then that keep the old filter still works. After all the filter change to the new world, then I can move this into the constructor.

Reply:
I'm confused. You call intialize() immediately after the constructor. Can't you just move this code into the constructor now?

Reply:
sorry, I didn't explain clear. In next version, I will initialize and reset file event for each filter iteration(move the initialize and reset inside the loop (https://github.com/envoyproxy/envoy/pull/17395/files#diff-6708370e112a5097f06e16009c3730b34f5ab0ee4f1e736f8816848409435888R104)

Then I can convert each filter one by one, and keep the existing filter(those register their own file event) works as before.

After all filters convert to the new way, then I will put the initialize() into the contract

Reply:
I think it would be fine to just fix all the filters in this PR and not need an intermediate step.

Reply:
got it